src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
-}
+  const cloneOfStates = [];
+  const cloneState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+
+        Object.assign(cloneState, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete cloneState[key];
+        }
+        break;
 
+      case 'clear':
+        for (const key of Object.keys(cloneState)) {
+          delete cloneState[key];
+        }
+        break;
```

Comment:
emm... why did you delete `default` case? You could just set a logic what to do when you've got an unexpected action
```suggestion
        break;
      
      default:
        throw new Error('Unexpected action!');
```

Reply:
добавив, я просто не знав як добавляти такі `Error`